Move all icons to the `@ckeditor/ckeditor5-icons` package

filipsobol opened the pull request:
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature: Move all icons to the `@ckeditor/ckeditor5-icons` package. Related to #16546.

Feature: Add `@ckeditor/ckeditor5-icons` package to the core DLL package.

MINOR BREAKING CHANGE: Move all icons to the `@ckeditor/ckeditor5-icons` package.

Removed the `twitter.svg` icon.
Renamed the following icons to `kebab-case` or to avoid name conflicts:
* `arrow-down.svg` (previously used in `revision-history`) to `chevron-down.svg`.
* `arrow-up.svg` (previously used in `revision-history`) to `chevron-up.svg`.
* `link.svg` (previously used in `uploadcare`) to `uploadcare-link.svg`.
* `image-upload.svg` (previously used in `uploadcare`) to `uploadcare-image-upload.svg`.
* `bookmark_inline.svg` to `bookmark-inline.svg`
* `bulletedlist.svg` to `bulleted-list.svg`
* `codeblock.svg` to `code-block.svg`
* `contentunlock.svg` to `content-unlock.svg`
* `exportpdf.svg` to `export-pdf.svg`
* `exportword.svg` to `export-word.svg`
* `horizontalline.svg` to `horizontal-line.svg`
* `importexport.svg` to `import-export.svg`
* `importword.svg` to `import-word.svg`
* `liststylecircle.svg` to `list-style-circle.svg`
* `liststyledecimalleadingzero.svg` to `list-style-decimal-leading-zero.svg`
* `liststyledecimal.svg` to `list-style-decimal.svg`
* `liststyledisc.svg` to `list-style-disc.svg`
* `liststylelowerlatin.svg` to `list-style-lower-latin.svg`
* `liststylelowerroman.svg` to `list-style-lower-roman.svg`
* `liststylesquare.svg` to `list-style-square.svg`
* `liststyleupperlatin.svg` to `list-style-upper-latin.svg`
* `liststyleupperroman.svg` to `list-style-upper-roman.svg`
* `numberedlist.svg` to `numbered-list.svg`
* `pagebreak.svg` to `page-break.svg`
* `specialcharacters.svg` to `special-characters.svg`
* `todolist.svg` to `todo-list.svg`

psmyrek reviewed (commented):
I found a few places with old paths to icons:

1. In `docs/framework/contributing/code-style.md`:

   ```
   import BoldIcon from '@ckeditor/ckeditor5-core/theme/icons/bold.svg';
   ```

2. In `docs/tutorials/supporting-multiple-versions.md`:

   ```
   import TableRowIcon from '@ckeditor/ckeditor5-table/theme/icons/table-row.svg';
   ```

3. In `docs/examples/custom/bottom-toolbar-editor.md`:

   ```
   import fontColorIcon from '@ckeditor/ckeditor5-font/theme/icons/font-color.svg';
   ```

In the `docs/updating/update-to-40.md` guide, paths to icons have been also changed in the sections describing old updates (to v40.1.0 and to v40.2.0). The new `ckeditor5-icons` package does not exist there yet.

Also, in the following files I found the webpack config `test: /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/` that probably needs to be aligned with new changes:

- `docs/getting-started/advanced/integrating-from-source-webpack.md`
- `docs/getting-started/legacy-getting-started/integrations/react.md`
- `docs/getting-started/legacy-getting-started/integrations/vuejs-v2.md`
- `docs/getting-started/legacy-getting-started/integrations/vuejs-v3.md`
- `docs/getting-started/legacy-getting-started/quick-start-other.md`
- `docs/updating/nim-migration/migration-to-new-installation-methods.md`

filipsobol commented:
I updated all docs except `docs/updating/update-to-40.md` because it's specific to an older version of the editor. Someone upgrading from `v39` to `v40` would get errors if we updated the guide to the solution from `v45`.

psmyrek commented on docs/examples/custom/bottom-toolbar-editor.md:
```diff
@@ -59,10 +59,12 @@ import {
 import { EasyImage } from 'ckeditor5-premium-features';
-import fontColorIcon from '@ckeditor/ckeditor5-font/theme/icons/font-color.svg';
+
+const fontColorIcon =/* #__PURE__ */ registerIcon( 'fontColor', IconFontColor );
```
`registerIcon()` is not available here yet.

psmyrek commented on docs/getting-started/advanced/integrating-from-source-webpack.md:
```diff
@@ -93,7 +93,7 @@ module.exports = {
 		rules: [
 			{
-				test: /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/,
+				test: /ckeditor5-icons\/theme\/icons\/[^/\\]+\.svg$/,
```
It looks like there are more problems with this regexp than expected 🥹 

Previous regexp with `[^/\\]` is invalid - unescaped `/` delimiter.
Current regexp also contains the same invalid `[^/\\]` fragment.

Also, for some reasons we were defining these regexps with both `\` and `/` path separators. Not sure if we can just support unix-style `/` now.

psmyrek replied:
> Previous regexp with `[^/\\]` is invalid - unescaped `/` delimiter.

Not true, sorry. My mistake, because when testing this regexp I chose PCRE2, not JS flavor. The regexp fragment `[^/\\]` is valid in JS.

So question is only whether we want to use only unix-style `/` path separator.

filipsobol replied:
How about we just go with `/\.svg$/`? I don't expect it to cause any more issues compared to the current version.

psmyrek replied:
Yes, good idea.

filipsobol replied:
Updated

psmyrek commented on docs/getting-started/advanced/integrating-from-source-webpack.md:
```diff
@@ -170,7 +170,7 @@ module.exports = {
 			},
 			{
-				test: /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/,
+				test: /cckeditor5-icons\/theme\/icons\/[^/\\]+\.svg$/,
```
`cckeditor5`

psmyrek reviewed (commented):
The old `/ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/` is still used in a few files.

![obraz](https://github.com/user-attachments/assets/11a42fd8-9a91-47f6-b81c-973e7a826012)

psmyrek reviewed (dismissed):
LGTM 🎉

filipsobol reviewed (commented):
It looks like we no longer use the `twitter.svg`, so it can be removed

psmyrek commented on docs/getting-started/setup/customizing-icons.md:
@godai78 Could you please scan this new guide to check if it requires any polishing?

filipsobol replied:
I updated the guide to also show how to override icons using Vite plugin

psmyrek reviewed (approved):
LGTM 🎉

godai78 reviewed (approved):
LGTM

psmyrek reviewed (approved):
LGTM 🎉
